tests/components/deconz/test_init.py

```diff
@@ -3,49 +3,51 @@
 import asyncio
 from unittest.mock import patch
 
-from homeassistant.components.deconz import (
-    DeconzHub,
-    async_setup_entry,
-    async_unload_entry,
+import pydeconz
+import pytest
+
+from homeassistant.components.deconz.const import (
+    CONF_MASTER_GATEWAY,
+    DOMAIN as DECONZ_DOMAIN,
 )
-from homeassistant.components.deconz.const import DOMAIN as DECONZ_DOMAIN
-from homeassistant.components.deconz.errors import AuthenticationRequired, CannotConnect
+from homeassistant.components.deconz.errors import AuthenticationRequired
+from homeassistant.config_entries import ConfigEntryState
 from homeassistant.core import HomeAssistant
 
 from .conftest import ConfigEntryFactoryType
 
 from tests.common import MockConfigEntry
 
 
-async def setup_entry(hass: HomeAssistant, entry: MockConfigEntry) -> None:
-    """Test that setup entry works."""
-    with (
-        patch.object(DeconzHub, "async_setup", return_value=True),
-        patch.object(DeconzHub, "async_update_device_registry", return_value=True),
-    ):
-        assert await async_setup_entry(hass, entry) is True
-
-
-async def test_setup_entry_successful(
-    hass: HomeAssistant, config_entry_setup: MockConfigEntry
-) -> None:
-    """Test setup entry is successful."""
-    assert hass.data[DECONZ_DOMAIN]
-    assert config_entry_setup.entry_id in hass.data[DECONZ_DOMAIN]
-    assert hass.data[DECONZ_DOMAIN][config_entry_setup.entry_id].master
+async def test_setup_entry(config_entry_setup: MockConfigEntry) -> None:
+    """Test successful setup of entry."""
+    assert config_entry_setup.state is ConfigEntryState.LOADED
+    assert config_entry_setup.options[CONF_MASTER_GATEWAY] is True
 
 
-async def test_setup_entry_fails_config_entry_not_ready(
-    hass: HomeAssistant, config_entry_factory: ConfigEntryFactoryType
+@pytest.mark.parametrize(
+    ("side_effect", "state"),
+    [
+        # Failed authentication trigger a reauthentication flow
+        (pydeconz.Unauthorized, ConfigEntryState.SETUP_ERROR),
+        # Connection fails
+        (TimeoutError, ConfigEntryState.SETUP_RETRY),
+        (pydeconz.RequestError, ConfigEntryState.SETUP_RETRY),
+        (pydeconz.ResponseError, ConfigEntryState.SETUP_RETRY),
+    ],
+)
+async def test_get_deconz_api_fails(
+    hass: HomeAssistant,
+    config_entry: MockConfigEntry,
+    side_effect: Exception,
+    state: ConfigEntryState,
 ) -> None:
-    """Failed authentication trigger a reauthentication flow."""
-    with patch(
-        "homeassistant.components.deconz.get_deconz_api",
-        side_effect=CannotConnect,
-    ):
-        await config_entry_factory()
-
-    assert hass.data[DECONZ_DOMAIN] == {}
+    """Failed setup."""
+    config_entry.add_to_hass(hass)
+    with patch("pydeconz.DeconzSession.refresh_state", side_effect=side_effect):
```

Comment:
We should patch where we use this

Reply:
You mean `homeassistant.components.deconz.api.something`?

Reply:
Exactly

Reply:
Fixed